template.go

```diff
@@ -48,9 +48,9 @@ func copyFuncMap(funcMap template.FuncMap) template.FuncMap {
 }
 
 // parseTemplate - parses text as a Go template with the given name and options
-func parseTemplate(ctx context.Context, name, text string, funcs template.FuncMap, tmplctx interface{}, nested config.Templates, leftDelim, rightDelim string) (tmpl *template.Template, err error) {
+func parseTemplate(ctx context.Context, name, text string, funcs template.FuncMap, tmplctx interface{}, nested config.Templates, leftDelim, rightDelim string, missingKey string) (tmpl *template.Template, err error) {
 	tmpl = template.New(name)
-	tmpl.Option("missingkey=error")
+	tmpl.Option("missingkey=" + missingKey)
```

Comment:
I think some validation of `missingKey` would be useful, throwing an error if the value is anything other than `error`, `zero`, or `default`.

Reply:
I wrote this before I saw the validation already done in `configfile.go` 😅... However I think it's still necessary here, because this could get called programmatically (through `RenderTemplates`).

Reply:
I have added the additional check here